ci: bump ci builder `0.50.0`

tynes opened the pull request:
**Description**

Bumps the `ci-builder` image with an updated version
of foundry.

Foundry update:
https://github.com/ethereum-optimism/optimism/pull/11325

Docker build:
https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/60263/workflows/154fd94a-4029-4e2f-8bce-4ea4c6d25897

<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

tynes committed:
ci: bump ci builder `0.50.0`

Bumps the `ci-builder` image with an updated version
of foundry.

Foundry update:
https://github.com/ethereum-optimism/optimism/pull/11325

Docker build:
https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/60263/workflows/154fd94a-4029-4e2f-8bce-4ea4c6d25897

tynes committed:
contracts-bedrock: fix tests after new foundry version

Inphi commented on packages/contracts-bedrock/test/L1/ResourceMetering.t.sol:
```diff
@@ -282,6 +282,8 @@ contract ArtifactResourceMetering_Test is Test {
     /// @dev Generates a CSV file. No more than the L1 block gas limit should
     ///      be supplied to the `meter` function to avoid long execution time.
     function test_meter_generateArtifact_succeeds() external {
+        vm.skip({ skipTest: true });
```
is this temporary? We should add a todo if so.

tynes replied:
Good call @Inphi, i have added a comment describing why this was added

tynes committed:
contracts-bedrock: comment why skip

mds1 commented on packages/contracts-bedrock/test/libraries/Bytes.t.sol:
```diff
@@ -68,6 +68,9 @@ contract Bytes_slice_Test is Test {
     /// @notice Tests that the `slice` function correctly updates the free memory pointer depending
     ///         on the length of the slice.
     function testFuzz_slice_memorySafety_succeeds(bytes memory _input, uint256 _start, uint256 _length) public {
+        _start = bound(_start, 0, _input.length - 1);
+        _length = bound(_length, 0, _input.length - _start);
+
         // The start should never be more than the length of the input bytes array - 1
         vm.assume(_start < _input.length);
         // The length should never be more than the length of the input bytes array - the starting
```
We should be able to remove both `vm.assume`s here as they are now no-ops

tynes replied:
I left them out of being worried that the `bound` call was not exactly correct

mds1 commented on packages/contracts-bedrock/test/libraries/Bytes.t.sol:
```diff
@@ -145,6 +148,9 @@ contract Bytes_slice_TestFail is Test {
     /// @notice Tests that, when given a start index `n` that is greater than
     ///         `type(uint256).max - n`, the `slice` function reverts.
     function testFuzz_slice_rangeOverflows_reverts(bytes memory _input, uint256 _start, uint256 _length) public {
+        _length = bound(_length, 0, _input.length);
+        _start = bound(_start, type(uint256).max - _length, type(uint256).max);
```
Since `bound` ranges are inclusive, to match the assume conditions that were here before I think this needs to be
```suggestion
        _length = bound(_length, 0, _input.length - 1);
        _start = bound(_start, type(uint256).max - _length + 1, type(uint256).max);
```

We can then similarly delete the assume comments (and perhaps preserve their comments by moving them above the corresponding bound statements)

tynes replied:
I did specifically not subtract 1 because it will panic when the input length is 0

tynes replied:
Ok so if we want to remove the `assume` calls, then we can do a `if length == 0 then 0 else length - 1` ?

mds1 replied:
Your bounds are looser than the `assume` bounds, and this avoids needing a conditional like `_input.length == 0 ? 0 : _input.length - 1`. So since we still have the `assume` statements below to catch the looser bounds, this is ok. I would suggest just adding a comment about this, since from the existing comments it's not clear which ranges (the bound ones vs. assume ones) are the desired ones

tynes replied:
This is a great comment, I will update

tynes commented:
more flakes

```
Failing tests:
Encountered 1 failing test in test/libraries/Bytes.t.sol:Bytes_slice_Test
[FAIL. Reason: assertion failed: 704 != 712; counterexample: calldata=0x84932af8000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000d500000000000000000000000000000000000000158bf9c597079cf4a0c63351ee000000000000000000000000000000000000000000000000000000000000000c96d6c4d81dd07f723593c11b0000000000000000000000000000000000000000 args=[0x96d6c4d81dd07f723593c11b, 213, 1707111784786006159357908308462 [1.707e30]]] testFuzz_slice_memorySafety_succeeds(bytes,uint256,uint256) (runs: 0, μ: 0, ~: 0)

Encountered 2 failing tests in test/libraries/SafeCall.t.sol:SafeCall_Test
[FAIL. Reason: send not successful; counterexample: calldata=0x491205050000000000000000000000000000000000000000000000000000000000001c760000000000000000000000005f65cd7d792e9746ef82929d60de9a1c526f93a500000000000000000000000000000000000000000000000000000000000012a000000000000000000000000000000000000000000000000000000000000018b4 args=[0x0000000000000000000000000000000000001c76, 0x5F65cD7D792E9746EF82929D60de9a1C526f93A5, 4768, 6324]] testFuzz_sendWithGas_succeeds(address,address,uint64,uint256) (runs: 8, μ: 51574, ~: 51574)
[FAIL. Reason: send not successful; counterexample: calldata=0x33717fe30000000000000000000000000000000000000000000000000000000000001c760000000000000000000000005f65cd7d792e9746ef82929d60de9a1c526f93a500000000000000000000000000000000000000000000000000000000000012a0 args=[0x0000000000000000000000000000000000001c76, 0x5F65cD7D792E9746EF82929D60de9a1C526f93A5, 4768]] testFuzz_send_succeeds(address,address,uint256) (runs: 1, μ: 51113, ~: 51113)
```

tynes committed:
test: better comments

tynes committed:
gas-snapshot: regenerate

tynes committed:
snapshots: update

tynes committed:
snapshots: update

tynes commented:
```
Failing tests:
Encountered 1 failing test in test/libraries/Bytes.t.sol:Bytes_slice_Test
[FAIL. Reason: assertion failed: 704 != 712; counterexample: calldata=0x84932af80000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000074adebfc81b36ef000000000000ea1d9a5ece9fc4626e42ca82665a5727203ba8458ded6f94323800000000000000000000000000000000000000000000000000000000000000191edfdc755aea79b17fe78c44c7ad7f8ea56822ea44fbbeaf5300000000000000 args=[0x1edfdc755aea79b17fe78c44c7ad7f8ea56822ea44fbbeaf53, 525477221819430639 [5.254e17], 376209323456661184620101705380240593498331650388362431821001272 [3.762e62]]] testFuzz_slice_memorySafety_succeeds(bytes,uint256,uint256) (runs: 0, μ: 0, ~: 0)

```

clabby committed:
fix(ctb): Align expected ptr in `Bytes.slice` test

An update to `forge-std` causes `bound` to set the free memory pointer
at an unaligned offset. This commit updates the test such that the
expected pointer is correctly aligned, without an assumption on the
starting ptr's alignment.
